packages/FamilyTree/Reflection/FamilyRelationsAnalyzer.php

```diff
@@ -36,4 +64,100 @@ public function getChildrenOfClassReflection(ClassReflection $desiredClassReflec
 
         return $childrenClassReflections;
     }
+
+    /**
+     * @param Name|NullableType|PhpParserUnionType|null $propertyTypeNode
+     */
+    public function getPossibleUnionPropertyType(
+        Property $property,
+        Type $varType,
+        ?Scope $scope,
+        ?Node $propertyTypeNode
+    ): PropertyType {
+        if ($varType instanceof UnionType) {
+            return new PropertyType($varType, $propertyTypeNode);
+        }
+
+        if (! $scope instanceof Scope) {
+            return new PropertyType($varType, $propertyTypeNode);
+        }
+
+        /** @var ClassReflection $classReflection */
+        $classReflection = $scope->getClassReflection();
+        $ancestors = $classReflection->getAncestors();
+        $propertyName = $this->nodeNameResolver->getName($property);
+
+        $kindPropertyFetch = $property->isStatic()
+            ? StaticPropertyFetch::class
+            : PropertyFetch::class;
+
+        $className = $property->getAttribute(AttributeKey::CLASS_NAME);
+        foreach ($ancestors as $ancestor) {
+            $ancestorName = $ancestor->getName();
+            if ($ancestorName === $className) {
+                continue;
+            }
+
+            $fileName = $ancestor->getFileName();
+            if ($fileName === false) {
+                continue;
+            }
+            $fileContent = $this->smartFileSystem->readFile($fileName);
+            $nodes = $this->parser->parse($fileContent);
+
+            $objectType = new ObjectType($ancestor->getName());
+            $parentType = new ObjectType('PHPUnit\Framework\TestCase');
+            $isPHPUnitTest = $parentType->isSuperTypeOf($objectType)
+                ->yes();
+
+            if ($isPHPUnitTest) {
+                continue;
+            }
+
+            if ($this->isFilled((array) $nodes, $propertyName, $kindPropertyFetch)) {
```

Comment:
Here the retype to `(array)` might cause similar troubles as `false` filename above. Maybe it would be better to handle `null` explicitly.

<br>

```suggestion
            if ($this->isFilled($nodes, $propertyName, $kindPropertyFetch)) {
```

Reply:
implemented https://github.com/rectorphp/rector-src/pull/295/commits/f65fa7eb18a93ec65bf649569e54760b88177352